rotkehlchen/db/accounting_rules.py

```diff
@@ -336,3 +303,151 @@ def query_rules_and_serialize(
+        events: Sequence[HistoryBaseEntry],
+) -> list[bool]:
+    """
+    For a list of events returns a list of the same length with boolean values where True
+    means that the event won't be affected by any accounting rule
+    """
+    query = 'SELECT COUNT(*) FROM accounting_rules WHERE (type=? AND subtype=? AND (counterparty=? OR counterparty=?))'  # noqa: E501
+    bindings = [
+        (
+            event.event_type.serialize(),
+            event.event_subtype.serialize(),
+            NO_ACCOUNTING_COUNTERPARTY if (counterparty := getattr(event, 'counterparty', None)) is None else counterparty,  # noqa: E501
+            NO_ACCOUNTING_COUNTERPARTY,  # account for rules based only on type/subtype
+        ) for event in events
+    ]
+    missing_accounting_rule: list[bool] = []
+    history_db = DBHistoryEvents(db)
+    callbacks = evm_accounting_aggregator.get_accounting_callbacks()
+    with db.conn.read_ctx() as cursor:
+        for idx, event in enumerate(events):
+            if isinstance(event, EthStakingEvent) is True:
+                # staking events all have a process() function for accounting
+                missing_accounting_rule.append(False)
+
+            # check the rule in the database
+            row = cursor.execute(query, bindings[idx]).fetchone()
+
+            if (is_missing_rule := row[0] == 0) is False:
+                missing_accounting_rule.append(is_missing_rule)
+                continue
+
+            if (  # check for cases that could use accounting_treatment
+                event.event_type == HistoryEventType.TRADE and
+                event.event_subtype in (HistoryEventSubType.RECEIVE, HistoryEventSubType.FEE)
+            ):
+                cursor.execute(
+                    query + ' AND (accounting_treatment=? OR accounting_treatment=?)',
+                    (
+                        HistoryEventType.TRADE.serialize(),
+                        HistoryEventSubType.SPEND.serialize(),
+                        NO_ACCOUNTING_COUNTERPARTY if (counterparty := getattr(event, 'counterparty', None)) is None else counterparty,  # noqa: E501
+                        NO_ACCOUNTING_COUNTERPARTY,
+                        TxAccountingTreatment.SWAP.serialize_for_db(),
+                        TxAccountingTreatment.SWAP_WITH_FEE.serialize_for_db(),
+
+                    ),
+                )
+                missing_accounting_rule.append(cursor.fetchone()[0] == 0)
+                continue
+
+            # check the surrounding events to verify if this particular event will
+            # be processed by a previous one. First query the related events
+            related_events = history_db.get_history_events(
+                cursor=cursor,
+                filter_query=HistoryEventFilterQuery.make(event_identifiers=[event.event_identifier]),
+                has_premium=True,
+            )
+
+            event_has_treatment = _check_if_event_affected_by_treatment(
```

Comment:
As I pointed out above, this should also return a number of event to skip. If it needs to skip events then in this for loop you should just skip ahead. Probably best way to do that is turn the for loop into an iterator and conume it.